engine/hatchery/main.go

```diff
@@ -59,9 +60,18 @@ func addCommands() {
 	rootCmd.AddCommand(openstack.Cmd)
 }
 
+// Cannot rely on viper.AutomaticEnv here because of the presence of hyphen '-'
+// in the flag (ie. for openstack cmd).
+// AutomaticEnv will only lookup for a strings.ToUpper
+// which will not replace hypen '-' with underscore '_'.
+// i.e "openstack-tenant" -> "CDS_OPENSTACK-TENANT" instead of "CDS_OPENSTACK_TENANT"
+// Hence SetEnvKeyReplacer.
+// Sadly, this isn't used by Flags().*Var functions, so no automatic binding on variables.
 func addFlags() {
```

Comment:
Oh nice ! i was stuck with it too !
is there a plan in viper to support/fix it ?